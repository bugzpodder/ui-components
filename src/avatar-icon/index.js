// @flow
export { AvatarIcon } from "./avatar-icon";
