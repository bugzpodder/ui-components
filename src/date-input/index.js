// @flow
export { wrapPickerUtilProvider } from "./picker-util-provider-hoc";
export { DateInput } from "./date-input";
export { DateTimeInput } from "./date-time-input";
