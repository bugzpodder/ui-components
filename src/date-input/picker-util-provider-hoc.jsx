// @flow
import MomentUtils from "material-ui-pickers/utils/moment-utils";
import MuiPickersUtilsProvider from "material-ui-pickers/utils/MuiPickersUtilsProvider";
import React from "react";

export function wrapPickerUtilProvider<P>(WrappedComponent: React$ComponentType<P>): React$ComponentType<P> {
  return (props: P) => {
    return (
      <MuiPickersUtilsProvider utils={MomentUtils}>
        <WrappedComponent {...props} />
      </MuiPickersUtilsProvider>
    );
  };
}
