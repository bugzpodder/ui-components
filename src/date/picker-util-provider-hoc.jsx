// @flow
import MomentUtils from "@date-io/moment";
import React from "react";
import { MuiPickersUtilsProvider } from "@material-ui/pickers";

export function wrapPickerUtilProvider<P>(WrappedComponent: React$ComponentType<P>) {
  return (props: P) => {
    return (
      <MuiPickersUtilsProvider utils={MomentUtils}>
        <WrappedComponent {...props} />
      </MuiPickersUtilsProvider>
    );
  };
}
