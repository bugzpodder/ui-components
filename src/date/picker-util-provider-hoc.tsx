import MomentUtils from "@date-io/moment";
import React, { ComponentType, useContext } from "react";
import {
  DATE_FORMAT,
  DATE_TIME_FORMAT,
  DATE_TIME_UNICODE_FORMAT,
  DATE_UNICODE_FORMAT,
} from "@grailbio/lib";
import {
  MuiPickersContext,
  MuiPickersUtilsProvider,
} from "@material-ui/pickers";
import {
  MuiPickersContext as OldMuiPickersContext,
  MuiPickersUtilsProvider as OldMuiPickersUtilsProvider,
} from "material-ui-pickers";
import { ParsableDate } from "@material-ui/pickers/constants/prop-types";

export function wrapPickerUtilProvider<P>(
  WrappedComponent: ComponentType<P>,
  useOldPicker = false,
  utils = MomentUtils,
) {
  return (props: P) => {
    return useOldPicker ? (
      <OldMuiPickersUtilsProvider utils={utils}>
        <WrappedComponent {...props} />
      </OldMuiPickersUtilsProvider>
    ) : (
      <MuiPickersUtilsProvider utils={utils}>
        <WrappedComponent {...props} />
      </MuiPickersUtilsProvider>
    );
  };
}

export const useMuiPickersContext = () => {
  const utils = useContext(MuiPickersContext);
  const oldUtils = useContext(OldMuiPickersContext);

  return utils || oldUtils;
};

export const useDateFormat = () => {
  const dateUtils = useMuiPickersContext();
  if (
    dateUtils &&
    dateUtils.yearFormat &&
    DATE_UNICODE_FORMAT.indexOf(dateUtils.yearFormat) >= 0
  ) {
    return {
      defaultDateFormat: DATE_UNICODE_FORMAT,
      defaultDateTimeFormat: DATE_TIME_UNICODE_FORMAT,
    };
  }

  return {
    defaultDateFormat: DATE_FORMAT,
    defaultDateTimeFormat: DATE_TIME_FORMAT,
  };
};

export const useFormattedDateForDisplay = (
  value: ParsableDate,
  format: string,
): string => {
  const dateUtils = useMuiPickersContext();
  const date = dateUtils ? dateUtils.date(value) : value;
  if (value && dateUtils && dateUtils.isValid(date)) {
    return dateUtils.format(date, format);
  }
  return "";
};
