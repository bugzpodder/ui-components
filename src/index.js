// @flow
import type { PagedTableProps } from "./table";

export { Alert } from "./alert";
export { AvatarIcon } from "./avatar-icon";
export { CommonCard } from "./common-card";
export { BreastCancerRibbon } from "./logos";
export { CommonDialog, CommonMultiPageDialog } from "./common-dialog";
export { CommonDropdownMenu } from "./common-dropdown";
export { CommonPanel } from "./common-panel";
export { CommonMultiSelect, CommonSelect } from "./common-select";
export { CommonSuggest, SuggestionsOmniField } from "./common-suggest";
export { CommonSwitch } from "./common-switch";
export {
  CommonPage, CommonTabbedPage, CommonPageV2, CommonTabbedPageV2,
} from "./page";
export {
  DateValue,
  DateTimeValue,
  HumanizedDateTime,
  DateInput,
  DateTimeInput,
  wrapPickerUtilProvider,
  TimeInput,
} from "./date";
export { ExternalLink, ExternalLinkButton, LinkButton } from "./link";
export { Navbar } from "./__deprecated__/navbar";
export { NavbarV2 } from "./navbar";
export { NotificationCenter } from "./notification-center";
export { ReadOnlyTextField, ValidatedReadOnlyTextField } from "./readonly-text-field";
export { OmniChips, OmniSearchBar } from "./omni-search-bar";
export {
  SimpleTable, ExportTableButton, LargeSimpleTable, PagedTable,
} from "./table";
export { ExportButton, generateReport } from "./export-button";
export type { PagedTableProps };
export { SignInPage } from "./sign-in-page";
export { SpinnerOverlay } from "./spinner-overlay";
export { StyleWrapper } from "./style-wrapper";
export { TabbedTimelineCard, TimelineGraph, TimelineCard } from "./timeline-graph";
export { TextInput, TextInputV2 } from "./text-input";
export { TwoColumnGrid, TwoColumnRow } from "./two-column-grid";
export { UploadButton } from "./upload-button";
export { SelectionGrid } from "./selection-grid";
export {
  // $FlowFixMe: Property `readFileListAsTextArray` is missing in ModuleNamespace for export {}
  readFileListAsTextArray,
  // $FlowFixMe: Property `readFileListAsText` is missing in ModuleNamespace for export {}
  readFileListAsText,
  // $FlowFixMe: Property `readFileAsText` is missing in ModuleNamespace for export {}
  readFileAsText,
} from "./utils";
