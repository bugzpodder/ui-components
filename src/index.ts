export * from "./alert";
export * from "./avatar-icon";
export * from "./common-card";
export { BreastCancerRibbon } from "./logos";
export { CommonDialog, CommonMultiPageDialog } from "./common-dialog";
export { CommonDropdownMenu } from "./common-dropdown";
export * from "./common-panel";
export * from "./common-select";
export * from "./common-suggest";
export * from "./common-switch";
export * from "./page";
export * from "./date";
export * from "./link";
export { Navbar } from "./__deprecated__/navbar";
export * from "./navbar";
export * from "./notification-center";
export * from "./readonly-text-field";
export * from "./omni-search-bar";
export * from "./table";
export * from "./export-button";
export * from "./sign-in-page";
export * from "./spinner-overlay";
export * from "./style-wrapper";
export * from "./timeline-graph";
export * from "./text-input";
export * from "./two-column-grid";
export * from "./upload-button";
export * from "./selection-grid";
export * from "./utils";
export * from "./types/base-navbar";
export * from "./types/dropdown";
export * from "./types/paged-table";
export * from "./types/selection-grid";
export * from "./types/timeline";
export * from "./types/card";
export * from "./types/grid";
export * from "./types/panel";
export * from "./types/switch";
export * from "./types/dialog";
export * from "./types/notification";
export * from "./types/select";
export * from "./types/table";
export { FavoriteIconButton } from "./dev";
