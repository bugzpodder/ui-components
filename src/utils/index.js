// @flow
export { COUNTRIES, SERGEY } from "./constants";
export { ExampleBlock } from "./example-block";
export { TestWrapper } from "./test-wrapper";
