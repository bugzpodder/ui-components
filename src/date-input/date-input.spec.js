// @flow
import React from "react";
import { render } from "react-testing-library";
import "jest-dom/extend-expect";
import { TestWrapper } from "@grail/components/src/utils";
import { wrapPickerUtilProvider } from "./picker-util-provider-hoc";
import { DateInput } from "./index";

const DateInputContainer = wrapPickerUtilProvider(DateInput);

test("render date input", async () => {
	const testDate = "2017-03-07 16:20:00";
	const mockCallback = jest.fn();
	const { container, getByPlaceholderText } = render(
		<TestWrapper>
			<DateInputContainer
				placeholder="Test Date Input"
				onChange={mockCallback}
				value={testDate} />
		</TestWrapper>,
	);
	expect(getByPlaceholderText("Test Date Input")).toHaveAttribute("value", "2017-03-07");
	expect(container).toMatchSnapshot();
});

test("render date input when readOnly is true", async () => {
	const testDate = "2017-03-07 16:20:00";
	const mockCallback = jest.fn();
	const { container, getByTestId } = render(
		<TestWrapper>
			<DateInputContainer
				placeholder="Test Date Input"
				onChange={mockCallback}
				value={testDate}
				readOnly={true} />
		</TestWrapper>,
	);
	expect(getByTestId("readonly-text-field")).toHaveTextContent("2017-03-07");
	expect(container).toMatchSnapshot();
});

test("render date input when readOnly is true showing - as empty value", async () => {
	const testDate = "";
	const mockCallback = jest.fn();
	const { container, getByTestId } = render(
		<TestWrapper>
			<DateInputContainer
				placeholder="Test Date Input"
				onChange={mockCallback}
				value={testDate}
				readOnly />
		</TestWrapper>,
	);
	expect(getByTestId("readonly-text-field")).toHaveTextContent("-");
	expect(container).toMatchSnapshot();
});

test("render date input when readOnly is true showing empty value", async () => {
	const testDate = "";
	const mockCallback = jest.fn();
	const { container, getByTestId } = render(
		<TestWrapper>
			<DateInputContainer
				placeholder="Test Date Input"
				onChange={mockCallback}
				value={testDate}
				readOnly
				showEmptyValue
			/>
		</TestWrapper>,
	);
	expect(getByTestId("readonly-text-field")).toHaveTextContent("");
	expect(container).toMatchSnapshot();
});
