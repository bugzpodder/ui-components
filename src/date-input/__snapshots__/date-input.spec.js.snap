// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render date input 1`] = `
<div>
   
  <div
    class="datePicker"
  >
    <div
      class="MuiFormControl-root"
    >
      <div
        class="MuiInput-root MuiInput-formControl MuiInput-underline WithUtils-DateTextField--input-1"
      >
        <input
          aria-invalid="false"
          class="MuiInput-input"
          placeholder="Test Date Input"
          type="text"
          value="2017-03-07"
        />
        <div
          class="MuiInputAdornment-root MuiInputAdornment-positionEnd adornmentWidth"
        >
          <button
            class="MuiButtonBase-root MuiIconButton-root"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiIconButton-label"
            >
              <span
                aria-hidden="true"
                class="material-icons MuiIcon-root"
              >
                 
                event
                 
              </span>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
      
    </div>
  </div>
   
</div>
`;

exports[`renderIntoDocument date input when readOnly is true 1`] = `
<div>
   
  <span
    class="readonlyTextField "
    data-testid="readonly-text-field"
  >
    2017-03-07
  </span>
   
</div>
`;

exports[`renderIntoDocument date input when readOnly is true showing - as empty value 1`] = `
<div>
   
  <span
    class="readonlyTextField "
    data-testid="readonly-text-field"
  >
    -
  </span>
   
</div>
`;

exports[`renderIntoDocument date input when readOnly is true showing empty value 1`] = `
<div>
   
  <span
    class="readonlyTextField "
    data-testid="readonly-text-field"
  >
    
  </span>
   
</div>
`;
